src/main/java/org/truffleruby/core/string/TBytesKey.java

```diff
@@ -12,19 +12,48 @@
 import java.util.Arrays;
 import java.util.Objects;
 
+import com.oracle.truffle.api.strings.InternalByteArray;
 import com.oracle.truffle.api.strings.TruffleString;
+import org.truffleruby.core.array.ArrayUtils;
 import org.truffleruby.core.encoding.RubyEncoding;
+import org.truffleruby.core.encoding.TStringUtils;
 
 public final class TBytesKey {
 
     private final byte[] bytes;
+    private final int offset;
+    private final int length;
+    private final boolean isImmutable;
```

Comment:
I think this should be tracked externally and passed to `makeCacheable()`, to save footprint.